core/src/test/scala/unit/kafka/server/KafkaApisTest.scala

```diff
@@ -3290,7 +3290,7 @@ class KafkaApisTest {
     when(clientQuotaManager.maybeRecordAndGetThrottleTimeMs(
       any[RequestChannel.Request](), anyDouble, anyLong)).thenReturn(0)
 
-    val fetchRequest = new FetchRequest.Builder(9, 9, -1, 100, 0, fetchDataBuilder)
+    val fetchRequest = new FetchRequest.Builder(9, 9, -1, -1, 100, 0, fetchDataBuilder)
```

Comment:
Should we extend tests here as well?

Reply:
This one does not have a direct way to verify how the following service handles the FetchRequestData. Also, on the replication side, we use the FetchRequest, which we have pretty good test coverage.